projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/graphql/GraphQLDataFetchers.java

```diff
@@ -176,7 +182,13 @@ private Criteria<V2DataJob> populateCriteria(List<Filter> filterList) {
                     f -> f,
                     filter ->
                         strategyFactory.findStrategy(
-                            JobFieldStrategyBy.field(filter.getProperty()))));
+                            JobFieldStrategyBy.field(filter.getProperty())),
+                    (v1, v2) -> {
+                      throw new IllegalArgumentException(
```

Comment:
What's the point of this exception? Won't it be thrown if there are duplicates?

Reply:
This is the current default behavior of the toMap function we use. However when supplying a map provider we need to also provide a duplicate handling function:
https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html#toMap-java.util.function.Function-java.util.function.Function-java.util.function.BinaryOperator-java.util.function.Supplier-

Current behavior:
https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html#toMap-java.util.function.Function-java.util.function.Function-

Which states in the docs that it throws IllegalStateException on duplicates.
So to summarize the only thing that we are changing is the map type provider to LinkedHashMap provider.